### DISCUSSION TITLE
Is it possible to import only some function from hardhat or plugins?

### DISCUSSION THREAD

**Discussion by ghost** [NomicFoundation/hardhat#3081] [Troubleshooting] [answered]
Is it possible to do smth like this using Typescript:
`import {getContractAt} from "hardhat/.../hardhat-ethers/..."`
or 
`import {name} from "hardhat/.../network/..."`
> *...* means some path

For example, I can do this:
`import {getContractAt} from "@nomiclabs/hardhat-ethers/internal/helpers"`
but it won't work, because it wants HRE, so I need to import this from hardhat (as I add this import to `hardhat-config.ts`).

And if so, will it take up less space or load faster?

**Comment by fvictorio** [comment, marked as answer]
In general you can't use any Hardhat functionality if the HRE (the Hardhat environment) is not loaded. We are going to work in the future on making Hardhat more modular (starting from being able to create a Hardhat network without having a Hardhat project), but things like `getContractAt` are very much tied to a Hardhat project.
